ngraph/frontend/onnx/frontend/include/onnx_frontend/place.hpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <frontend_manager/place.hpp>
+
+namespace ngraph
+{
+    namespace frontend
+    {
+        class FRONTEND_API PlaceInputEdgeONNX : public Place
+        {
+        public:
+            PlaceInputEdgeONNX(const onnx_editor::InputEdge& edge)
+                : m_edge(edge)
+            {
+            }
+
+        private:
+            onnx_editor::InputEdge m_edge;
+        };
+
+        class FRONTEND_API PlaceOutputEdgeONNX : public Place
+        {
+        public:
+            PlaceOutputEdgeONNX(const onnx_editor::OutputEdge& edge)
+                : m_edge(edge)
+            {
+            }
+
+        private:
+            onnx_editor::OutputEdge m_edge;
+        };
+
+        class FRONTEND_API PlaceTensorONNX : public Place
+        {
+        public:
+            PlaceTensorONNX(const std::string& name, const onnx_editor::ONNXModelEditor& editor)
+                : m_name(name)
+                , m_editor(editor)
+            {
+            }
+
+            std::vector<std::string> get_names() const override { return {m_name}; }
+
+            Place::Ptr get_producing_port() const override
+            {
+                return std::make_shared<PlaceOutputEdgeONNX>(m_editor.find_output_edge(m_name));
+            }
+
+            std::vector<Place::Ptr> get_consuming_ports() const override
+            {
+                std::vector<Place::Ptr> ret;
+                auto edges = m_editor.find_output_consumers(m_name);
+                std::transform(edges.begin(),
+                               edges.end(),
+                               std::back_inserter(ret),
+                               [](const onnx_editor::InputEdge& edge) {
+                                   return std::make_shared<PlaceInputEdgeONNX>(edge);
+                               });
+                return ret;
+            }
+
+            Ptr get_input_port(int input_port_index) const override
+            {
+                return std::make_shared<PlaceInputEdgeONNX>(m_editor.find_input_edge(
+                    onnx_editor::EditorNode(m_name), onnx_editor::EditorInput(input_port_index)));
+            }
+
+        private:
+            std::string m_name;
+            const onnx_editor::ONNXModelEditor& m_editor;
+        };
+    } // namespace frontend
```

Comment:
These classes are internal implementation details. Consider moving the entire file to src subdirectory. We haven't finally negotiated which part of API should be available at package level, and which is not. But for these classes it is more clear, they are internal parts of the FE implementation.

PDPD frontend doesn't follow the same approach at the moment, but it is subject to change. We are considering to expose in include directory of a frontend the following components only:

1. FrontEnd class for users who don't need dispatching and auto-loading service from FrontEndManager. They will include header with front end class directly and link to it directly.
2. Extensions for users who want to register their own operator convertors or transformations. Extension sub-system is currently under architecture development, but it will likely be expressed as add_extension method in FrontEnd class. It relies on FrontEnd Extension classes that are specific for each front-ends and should provide capabilities similar to onnx_imort register new operator capability. Necessary classes to implement such extension should be provided as a part of public API of FrontEnd.

So specializations for InputModel and various Places are considered as internals.